webapi/CopilotChat/Skills/ChatSkills/ExternalInformationSkill.cs

```diff
@@ -118,7 +118,25 @@ public class ExternalInformationSkill
         {
             // Create a plan and set it in context for approval.
             var contextString = string.Join("\n", context.Variables.Where(v => v.Key != "userIntent").Select(v => $"{v.Key}: {v.Value}"));
-            Plan plan = await this._planner.CreatePlanAsync($"Given the following context, accomplish the user intent.\nContext:\n{contextString}\nUser Intent:{userIntent}");
+            Plan? plan = null;
+            int maxRetries = 1;
```

Comment:
should this be controllable by the user? having the max set as 1 seems to defeat the purpose of the loop below.

Reply:
This should not be user-configurable since this retry is a temporary stabilization until the root causes are fixed in the Planner, so we need to heed any retry logic with warning. Aligned with Nilesh that we should limit this kind of retry to 1 so we can accurately identify SK failures.

Reply:
ok that makes sense. right now the fn will continue to retry until a valid plan is reached, no matter the value of `maxRetries`, is that intentional?

Reply:
It will throw and consequently bail out if it fails the retry

Reply:
I added the throw back in explicitly